test_parse_directory.py

```diff
@@ -37,19 +45,56 @@ def report_status(succeeded, file, verbose, error=None):
         print(f"  {str(error.__class__.__name__)}: {error}")
 
 
+def generate_parser(grammar_file):
```

Comment:
Can you factor this out into a utility that's also called from `__main__.py`? It doesn't need to set or return `endpos` -- we can just drop that from the output (it was more useful before I had a line count, to estimate the compilation speed).

The helper function could return the exception object representing the SyntaxError, or None for success.